app/scripts/modules/core/src/presentation/tables/Table.tsx

```diff
@@ -0,0 +1,67 @@
+import * as React from 'react';
+
+import { useIsMobile } from 'core/presentation';
+
+import { ITableRowLayoutProps } from './TableRow';
+import { ITableCellLayoutProps } from './TableCell';
+import { TableContext } from './TableContext';
+
+const { useState, useMemo } = React;
+
+export interface ITableColumn {
+  name: string;
+  sortDirection?: 'ascending' | 'descending';
+  onSort?: () => any;
+}
+
+export type ITableColumns = Readonly<Array<string | ITableColumn>>;
+
+export type ITableLayoutProps = Omit<ITableProps, 'layout' | 'columns'> & {
+  columns: ITableColumn[];
+  isMobile: boolean;
+  expandAll: () => any;
+};
+
+export interface ITableLayout {
+  getTableLayout: (props: ITableLayoutProps) => JSX.Element;
+  getTableRowLayout: (props: ITableRowLayoutProps) => JSX.Element;
+  getTableCellLayout: (props: ITableCellLayoutProps) => JSX.Element;
+}
+
+export interface ITableProps {
+  layout: ITableLayout;
+  columns: ITableColumns;
+  expandable?: boolean;
+  children?: React.ReactNode;
+}
+
+export const Table = (props: ITableProps) => {
+  const isMobile = useIsMobile();
+  const [allExpanded, setAllExpanded] = useState(false);
+
+  const { layout, ...propsWithoutLayout } = props;
+
+  const normalizedColumns = props.columns.map(column => (typeof column === 'string' ? { name: column } : column));
+  const tableContext = useMemo(
+    () => ({
+      layout,
+      isMobile,
+      allExpanded,
+      setAllExpanded: () => setAllExpanded(false),
+      expandable: props.expandable,
+      columns: normalizedColumns,
+    }),
+    [layout, isMobile, props.expandable, allExpanded, props.columns],
+  );
+
+  return (
+    <TableContext.Provider value={tableContext}>
+      {layout.getTableLayout({
```

Comment:
👍 yeah, I went back and forth on this a bit and in hindsight I think I agree. One nice freebie you get out of this is the ability to use hooks in layout code without having to render a component inside a component the way I did in `standardGridTableLayout` (in the current impl trying to use a hook in layout code would be weird because it's technically not at the top of the render function for the component it lives inside).